lib/src/dbus_ws_read_buffer.dart

```diff
@@ -0,0 +1,386 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'dbus_buffer.dart';
+import 'dbus_bus_name.dart';
+import 'dbus_error_name.dart';
+import 'dbus_interface_name.dart';
+import 'dbus_member_name.dart';
+import 'dbus_message.dart';
+import 'dbus_value.dart';
+
+/// Decodes DBus messages from binary data.
+class DBusWSReadBuffer {
+  /// Json data in the buffer.
+  String _responseData = "";
+
+  void setResponseData(String responsedata) {
+    _responseData = responsedata;
+  }
+
+  /// Reads a D-Bus message from the buffer or returns null if not enough data.
```

Comment:
it's not clear to me that this method will handle 'not enough data' scenario properly (probably json.decode would throw if the message is not complete?) - perhaps this comment ('returns null if not enough data') might be misleading

Reply:
done